### PR TITLE
create memory descriptors based on the tensor dimensions

### PR DESCRIPTION
**Pull request by snadampal**
Arm Compute Library(ACL)backend requires explicit memory format tag iniatilization to decide wether the tensor can be computed with the ACL kernels. Hence, the src, weights and dst memroy descriptor format is set based on the tensor dimensions instead of using the format::any tag.

### Description

Arm Compute Library(ACL)backend requires explicit memory format tag iniatilization to decide wether the tensor can be computed with the ACL kernels. Hence, the src, weights and dst memroy descriptor format is set based on the tensor dimensions instead of using the format::any tag.

### Motivation and Context

The change enables ACL kernels for DNNL matmul ops on aarch64 platform.

### CODE REVIEW

**Comment by snadampal** [comment]
Hi @tianleiwu , I've signed the CLA, please let me if there is any feedback or this PR can be merged. Thank you!

**Comment by snadampal** [comment]
Hi @tianleiwu , sorry for the delayed response. I have rebased the PR and also ran lintrunner -a.

File: onnxruntime/core/providers/dnnl/subgraph/dnnl_matmul.cc [hunk 1]
```diff
@@ -139,14 +158,14 @@ void DnnlMatMul::CreatePrimitive(DnnlSubgraphPrimitive& sp, DnnlNode& node) {
   if (transA || transBatchA) {
     src_md = transposedA_md;
   } else {
-    src_md = dnnl::memory::desc(src_dims, node.Input(IN_A).Type(), dnnl::memory::format_tag::any);
+    src_md = dnnl::memory::desc(src_dims, node.Input(IN_A).Type(),  get_default_format(src_dims));
```
Comment (tianleiwu) [inline onnxruntime/core/providers/dnnl/subgraph/dnnl_matmul.cc:161]: please run `lintrunner -a`. there is extra space after `,`

Comment (snadampal) [reply onnxruntime/core/providers/dnnl/subgraph/dnnl_matmul.cc]: thanks, i have pushed the update and checking how it was missed when i ran lintrunner -a on this version.

**Review by tianleiwu — APPROVED** [review]

**Pull request merged**